### PR TITLE
[CLD-6501] Enhance visibility for PR builds artefacts

### PR DESCRIPTION
**Pull request by toninis**
#### Summary

- Update Dockerfile by enforcing proper entrypoint ownership/permissions 
- Create separate workflow for Sentry upload
- Change naming convention for mattermost team edition packaging to `Server CI Artifacts` 
- Generate summary with links for the packages
- Create new status check on PR `Server CI/Artifacts Build` to enhance testing

> Currently all team edition images from [mattermostdevelopment/mm-te-test](https://hub.docker.com/r/mattermostdevelopment/mm-te-test/tags) are not working because of the way we create the docker images . Github by default strips all permissions when uploads an artifact with the **upload-artifact** action . This strips also the +x permission on entrypoint.sh . We resolve this issue with [this](https://github.com/mattermost/mattermost/commit/8aa642c8c7482196d55dba0c8945ecac61d24029) change .

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-6501

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by saturninoabril — APPROVED** [review]
Thanks @toninis, LGTM!

File: .github/workflows/server-ci-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    workflows:
+      - "Server CI PR"
+    types:
+      - completed
+
+jobs:
+  ## We only need the condition on the first job
+  ## This will run only when a pull request is created with server changes
+  update-initial-status:
+    if: github.repository_owner == 'mattermost' && github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
```
Comment (lieut-data) [inline .github/workflows/server-ci-artifacts.yml:14]: If I fork and edit this file to remove the `repository_owner` check, does anything bad happen when I push the PR?

File: server/build/Dockerfile [hunk 2]
```diff
@@ -39,7 +39,7 @@ HEALTHCHECK --interval=30s --timeout=10s \
   CMD curl -f http://localhost:8065/api/v4/system/ping || exit 1
 
 # Configure entrypoint and command
-COPY entrypoint.sh /
+COPY --chown=mattemrost:mattermost --chmod=665 entrypoint.sh /
```
Comment (lieut-data) [inline server/build/Dockerfile:42]: Alternatively, could we specify the entrypoint as `/bin/bash entrypoint.sh`?

**Review by lieut-data — APPROVED** [review]
Two questions, but no need for a re-review for me unless you need it :)

File: .github/workflows/server-ci-artifacts.yml (hunk 1 above)
Comment (phoinixgrr) [reply .github/workflows/server-ci-artifacts.yml:14]: @lieut-data  

Why deny inheritance of Secrets when running in forks. 
Moreover the GITHUB_TOKEN secret has read-only permissions in pull requests from forked repositories.

File: server/build/Dockerfile (hunk 2 above)
Comment (phoinixgrr) [reply server/build/Dockerfile:42]: There is also a typo in UID here 

```suggestion
COPY --chown=mattermost:mattermost --chmod=665 entrypoint.sh /
```

**Review by phoinixgrr — CHANGES_REQUESTED** [review]

**Review by phoinixgrr — APPROVED** [review]

**Pull request merged**